spec/Section 3 -- Type System.md

```diff
@@ -1340,14 +1357,281 @@ Union type extensions have the potential to be invalid if incorrectly defined.
 
 1. The named type must already be defined and must be a Union type.
 2. The member types of a Union type extension must all be Object base types;
-   Scalar, Interface and Union types must not be member types of a Union.
-   Similarly, wrapping types must not be member types of a Union.
+   Scalar, Interface, Union and Tagged types must not be member types of a
+   Union.  Similarly, wrapping types must not be member types of a Union.
 3. All member types of a Union type extension must be unique.
 4. All member types of a Union type extension must not already be a member of
    the original Union type.
 5. Any non-repeatable directives provided must not already apply to the
    original Union type.
 
+
+## Tagged Types
+
+TaggedTypeDefinition : Description? tagged Name Directives[Const]? TaggedMembersDefinition?
+
+TaggedTypeExtension :
+  - extend tagged Name Directives[Const]? TaggedMembersDefinition
+  - extend tagged Name Directives[Const]
+
+TaggedMembersDefinition : { TaggedMemberDefinition+ }
+
+TaggedMemberDefinition : Description? Name : Type Directives[Const]?
+
+Tagged types represent a list of possible named members, exactly one of which
```

Comment:
Curious for thoughts on the use of a new term "members" here vs just reusing "fields" again. "field" already has some ambiguous usage when used across input and output object types, so I can see both sides of the argument to reuse it or invent a new term.

Reply:
I think `member fields` makes most sense. They're field-like except that exactly one will be present so the nullability rules you know about input fields, for example, don't apply here. (A tagged input type could have 3 non-nullable fields, but you'd still only specify one of them.)

Reply:
I've moved to using `member fields` so am resolving this :+1: